client/login/controller.js

```diff
@@ -27,13 +27,19 @@ const enhanceContextWithLogin = context => {
 		query,
 	} = context;
 
+	const previousHash = context.state || {};
+	const { client_id, user_email, user_name, id_token, state } = previousHash;
+	const socialServiceResponse = client_id
+		? { client_id, user_email, user_name, id_token, state }
```

Comment:
why not just use `previousHash`

Reply:
Because previousHash might contain more properties that we don't want in our `socialServiceResponse`. Moreover, destructuring `previousHash` makes sure you'll get each property in `socialServiceResponse` even if it's not present in the hash object